sale_order_import_edifact/wizard/sale_order_import.py

```diff
@@ -0,0 +1,236 @@
+import json
+import logging
+import os
+from base64 import b64decode
+from collections import defaultdict
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+logger = logging.getLogger(__name__)
+
+
+class SaleOrderImportEdifact(models.TransientModel):
+    _name = "sale.order.import"
+    _inherit = ["sale.order.import", "base.edifact"]
+
+    import_type = fields.Selection(
+        selection_add=[("edifact", "EDIFACT")], ondelete={"edifact": "cascade"}
+    )
+    state = fields.Selection(
+        selection_add=[("tech", "Technical")], ondelete={"tech": "set default"}
+    )
+    parsed_json = fields.Text()
+
+    @property
+    def edifact_ok(self):
+        conf_ext = self.env.context.get("file_ext", "txt,d96a")
+        extensions = conf_ext.split(",")
+        ok = False
+        if self.order_filename:
+            path, ext = os.path.splitext(self.order_filename)
+            ok = ext and ext[1:] in extensions
+            if not ok and self.order_file:
+                ok = b64decode(self.order_file[:4]) == b"UNB"
+        return ok
+
+    @api.onchange("order_file")
+    def order_file_change(self):
+        if self.edifact_ok:
+            self.doc_type = self.env.context.get("doc_type", "rfq")
+            self.price_source = "order"
+        else:
+            res = super(SaleOrderImportEdifact, self).order_file_change()
+            if isinstance(res, dict):
+                return res
+
+    def parse_order_button(self):
+        self.ensure_one()
+        order_file_decoded = b64decode(self.order_file)
+        parsed_obj = dict(edifact=None, order=None)
+        if self.edifact_ok:
+            parsed_obj["edifact"] = self.pydifact_obj(order_file_decoded)
+
+        parsed_order = self.parse_order(
+            order_file_decoded, self.order_filename, self.partner_id
+        )
+        del parsed_order["attachments"]
+        parsed_obj["order"] = parsed_order
+        self.write(
+            {
+                "parsed_json": json.dumps(parsed_obj, indent=4, default=str),
+                "state": "tech",
+            }
+        )
+        action = self.env["ir.actions.act_window"]._for_xml_id(
+            "sale_order_import.sale_order_import_action"
+        )
+        action["res_id"] = self.id
+        return action
+
+    def _get_supported_types(self):
+        # Add more types for EDIFACT
+        res = super()._get_supported_types()
+        res.update({"edifact": ("text/plain")})
+        return res
+
+    @api.model
+    def parse_edifact_order(self, filecontent, detect_doc_type):
```

Comment:
Signature to respect
```suggestion
    def parse_edifact_order(self, filecontent, detect_doc_type=detect_doc_type):
```